src/add-search-spec.js

```diff
@@ -114,34 +125,36 @@ describe('Search Results', () => {
           'search'
         );
 
-        content['content'] = pageContents;
+        content.stachePageSearchData = pageContents;
 
         if (!fs.existsSync(searchDirPath)) {
           fs.mkdirSync(searchDirPath);
         }
         return writeSearchFile(searchDirPath);
       })
       .then(() => done())
-      .catch(error => {
+      .catch((error: any) => {
         console.log('ERROR', error);
         expect(error).toBeNull();
         done();
       });
   });
 });
-`
+`;
 
-function addSearchSpecToProject() {
-    try {
-        if (!fs.existsSync('e2e')) {
-            fs.mkdirSync('e2e');
+// http://web.archive.org/web/20161108071447/http://blog.osteele.com/posts/2007/12/cheap-monads/
+function addSearchSpecToProject(argv, config) {
+    if ((((config || {}).appSettings || {}).stache || {}).search) {
```

Comment:
You could put this in the `try` block, since calling this method without the appropriate config would constitute a failure:

```
try {
  if (config.appSettings.stache.search) {
    // Continue on...
  }
} catch() {}
```

Reply:
^ Reasoning: the `if` statement would be more readable.

Reply:
Well... it might not constitute a failure. Since it's part of the build process, that command will get called regardless. If the config is undefined, that would be an error, but if it is defined but false, it's not. Plus I'll have to account for undefined values regarless in there. It'll throw typeerrors if it's improperly structured. This syntax prevents that.